### DISCUSSION TITLE
Wasm DataGrid Row detail dissapearing

### DISCUSSION THREAD

**Discussion by DomagojJugovich** [unoplatform/uno#11172] [General]
Video is attached.
First half of video  is UWP head that works, second half WASM in Chrome that missbehaves, also resizing columns does nothing (not in video).

https://user-images.githubusercontent.com/3256682/216392189-c980e63e-e01e-417d-92f0-1285434ee966.mp4

**Comment by DomagojJugovich** [comment]
![image](https://user-images.githubusercontent.com/3256682/216392706-65ae2430-89b2-4171-bbb7-b9cf20b621e9.png)
![image](https://user-images.githubusercontent.com/3256682/216393060-1b7a6d91-09a6-4df5-8685-7971f4aaad66.png)

**Reply by jeromelaban** [reply]
It looks like something was not backported to the UWP version of the Community Toolkit for Uno. Can you try with the 7.1.10 version of the toolkit and let us know?

**Reply by DomagojJugovich** [reply]
It is the same with 7.1.10 in WASM 

(cleaned obj/bin folders to be shure, clean/rebuild)
(cleaned cache and site data in chrome)

**Reply by DomagojJugovich** [reply]
XAML of DataGrid

```xaml
<controls:DataGrid Grid.Column="0" Grid.Row="1" AutoGenerateColumns="False" Name="machinesGrid" CanUserResizeColumns="True" MaxColumnWidth="420"
                           CanUserSortColumns="True" AlternatingRowBackground="AliceBlue" RowHeight="68" 
                           RowDetailsVisibilityMode="VisibleWhenSelected">
            <controls:DataGrid.Columns>
                <controls:DataGridTextColumn Header="Machine" CanUserSort="True" 
                                           Binding="{Binding MachineName, Mode=OneWay}" IsReadOnly="True"/>
                <controls:DataGridTextColumn Header="IP" 
                                           Binding="{Binding IPadress, Mode=OneWay}" IsReadOnly="True"/>
                <controls:DataGridTextColumn Header="User" MinWidth="175" CanUserSort="True"
                                           Binding="{Binding UserName, Mode=OneWay}" IsReadOnly="True"/>
                 <controls:DataGridTemplateColumn Header="LastSync" >
                    <controls:DataGridTemplateColumn.CellTemplate>
                        <DataTemplate>
                            <Border Background="{Binding LastSync,Converter={StaticResource ValueToBrushConv}}">
                                <TextBlock Text="{Binding LastSync, Mode=OneWay}" ></TextBlock>
                            </Border>
                        </DataTemplate>
                    </controls:DataGridTemplateColumn.CellTemplate>
                </controls:DataGridTemplateColumn>
                <controls:DataGridTextColumn Header="AppFolderVersions" MinWidth="175"  CanUserSort="True"
                                           Binding="{Binding AppFolderVersions, Mode=OneWay}" IsReadOnly="True"/>
            </controls:DataGrid.Columns>
            <controls:DataGrid.RowDetailsTemplate>
                <DataTemplate x:DataType="local:Machine">
                    <Grid Margin="20,10" Padding="5" Height="500">
                        <Grid.ColumnDefinitions>
                            <ColumnDefinition Width="*" />
                        </Grid.ColumnDefinitions>
                        <Grid.RowDefinitions>
                            <RowDefinition Height="*" />
                            <RowDefinition Height="*" />
                        </Grid.RowDefinitions>
                        <TextBox TextWrapping="Wrap" Grid.Column="0" Grid.Row="0" IsReadOnly="True" Text="{x:Bind Path=LogChanges}" ScrollViewer.HorizontalScrollBarVisibility="Auto"
                                ScrollViewer.VerticalScrollBarVisibility="Auto" 
                                />
                        <TextBox TextWrapping="Wrap" Grid.Column="0" Grid.Row="1" IsReadOnly="True" Foreground="Red" FontSize="13" Text="{x:Bind Path=LogErrors}" ScrollViewer.HorizontalScrollBarVisibility="Auto"
                                ScrollViewer.VerticalScrollBarVisibility="Auto"
                                />
                   </Grid>
                 </DataTemplate>
            </controls:DataGrid.RowDetailsTemplate>
        </controls:DataGrid>
```

**Reply by jeromelaban** [reply]
Thanks for the report. Let's follow in https://github.com/unoplatform/uno/issues/11184.

Here's a repro in the Playground: https://playground.platform.uno/#6cc47a2d

**Comment by DomagojJugovich** [comment]
UWP project references ::::::


`   <PackageReference Include="Microsoft.NETCore.UniversalWindowsPlatform">
      
      <Version>6.2.11</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.Toolkit.Uwp">
      <Version>7.1.3</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.Toolkit.Uwp.UI">
      <Version>7.1.3</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.Toolkit.Uwp.UI.Controls">
      <Version>7.1.3</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.Toolkit.Uwp.UI.Controls.DataGrid">
      <Version>7.1.3</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.UI.Xaml" Version="2.7.1" />
    <PackageReference Include="System.Text.Json">
      <Version>7.0.1</Version>
    </PackageReference>
    <PackageReference Include="Uno.UI" Version="4.7.30" />
    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="5.0.0" />
    <PackageReference Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />`

**Comment by DomagojJugovich** [comment]
I also tried to make 4.7  newest template with WinUI, WASM (no UWP) but cant get WASM to work and gave up for now.
